### PR TITLE
[IMP] auditlog - Autovacuum logs

### PR DESCRIPTION
**Pull request by sebalix**
Add a `ir.cron` to delete logs periodically (fixing #187).

The index on the `log_id` field on the `auditlog_log_line` table is required to delete thousands of logs almost instantaneously.

I will bump the version number after the PR #357 is merged.

### CODE REVIEW

**Comment by hbrunn** [comment]
:+1: , but adding deletion to tests would be nice

**Comment by sebalix** [comment]
@hbrunn I will do that

**Comment by sebalix** [comment]
Thinking about it, I will rewrite this to purge user sessions and HTTP requests records too.

**Comment by sebalix** [comment]
Ready for review.

File: auditlog/README.rst [hunk 1]
```diff
@@ -11,6 +11,11 @@ Go to `Reporting / Audit / Rules` to subscribe rules. A rule defines which
 operations to log for a given data model.
```
Comment (elicoidal) [inline auditlog/README.rst:16]: Hi @sebalix Thanks for the contribution.
I miss the top of the README file: badge + line for document title.

File: auditlog/__openerp__.py [hunk 2]
```diff
@@ -21,7 +21,7 @@
 
 {
     'name': "Audit Log",
-    'version': "8.0.1.2.0",
+    'version': "8.0.1.3.0",
```
Comment (elicoidal) [inline auditlog/__openerp__.py:24]: According to https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#version-numbers I would tend to have 8.0.2.0.0  (new views and new data in models).
What do you think?

File: auditlog/data/ir_cron.xml [hunk 3]
```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0" encoding="utf-8"?>
+<openerp>
+    <data noupdate="1">
+
+        <record id="ir_cron_auditlog_autovacuum" model="ir.cron">
+            <field name='name'>AutoVacuum audit logs</field>
```
Comment (elicoidal) [inline auditlog/data/ir_cron.xml]: s/AutoVacuum/Auto-vacuum

File: auditlog/models/log.py [hunk 4]
```diff
@@ -18,7 +18,6 @@
 #    along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #
 ##############################################################################
-
```
Comment (elicoidal) [inline auditlog/models/log.py:21]: Can you take the opportunity to simplify all headers from py files?

File: auditlog/README.rst [hunk 5]
```diff
@@ -11,6 +11,11 @@ Go to `Reporting / Audit / Rules` to subscribe rules. A rule defines which
...
 Then, check logs in the `Reporting / Audit / Logs` menu.
 
+By default logs older than 6 months (180 days) will be deleted automatically.
+To change this delay, go to the
+`Configuration / Technical / Automation / Scheduled Actions` menu and edit the
+`AutoVacuum audit logs` task parameters.
+
 During installation, it will migrate any existing data from the `audittrail`
```
Comment (elicoidal) [inline auditlog/README.rst]: During installation, a one-time script will migrate any existing data from the `audittrail`
 module (rules and logs).

File: auditlog/README.rst [hunk 6]
```diff
@@ -11,6 +11,11 @@ Go to `Reporting / Audit / Rules` to subscribe rules. A rule defines which
...
 module (rules and logs).
```
Comment (elicoidal) [inline auditlog/README.rst]: Remove the part below on the forum, not necessary anymore.

Comment (elicoidal) [inline auditlog/README.rst]: in the url below: https (x2)

File: auditlog/__openerp__.py [hunk 7]
```diff
@@ -21,7 +21,7 @@
```
Comment (elicoidal) [inline auditlog/__openerp__.py:21]: simplify header

File: auditlog/__openerp__.py (hunk 2 above)
Comment (sebalix) [inline auditlog/__openerp__.py:24]: I checked this document, and as it is a non-breaking new feature which does not require a data migration, I opted for `1.3.0`. This feature is less intrusive than #357 where I have done the same incrementation. But it does not matter for me if it has to be `2.0.0`, what other peoples think?

Comment (elicoidal) [inline auditlog/__openerp__.py:24]: Finally not so sure: 8.0.1.3.0 looks good...

File: auditlog/data/ir_cron.xml (hunk 3 above)
Comment (sebalix) [inline auditlog/data/ir_cron.xml]: I just copied https://github.com/OCA/connector/blob/9.0/connector/queue/queue_data.xml#L31 to be identical with others and have a nice and coherent overview of the scheduled actions in the tree view:
- AutoVacuum audit logs
- AutoVacuum osv_memory objects
- AutoVacuum Queue Jobs

But your suggestion is right.

File: auditlog/models/log.py (hunk 4 above)
Comment (sebalix) [inline auditlog/models/log.py:21]: I agree, but I would like to do these kind of changes in another PR (simplify headers, add screenshots, improve/clean up the README...).

File: auditlog/data/ir_cron.xml (hunk 3 above)
Comment (elicoidal) [inline auditlog/data/ir_cron.xml]: Your call: this is not so important if it is consistent with existing "misspelling" ;)

File: auditlog/models/log.py (hunk 4 above)
Comment (elicoidal) [inline auditlog/models/log.py:21]: OK for me

File: auditlog/README.rst (hunk 5 above)
Comment (sebalix) [inline auditlog/README.rst]: Fixed.

File: auditlog/README.rst (hunk 6 above)
Comment (sebalix) [inline auditlog/README.rst]: Fixed.

File: auditlog/data/ir_cron.xml (hunk 3 above)
Comment (sebalix) [inline auditlog/data/ir_cron.xml]: So be it, fixed ;)

File: auditlog/README.rst (hunk 1 above)
Comment (sebalix) [inline auditlog/README.rst:16]: Fixed.

File: auditlog/README.rst [hunk 8]
```diff
@@ -1,5 +1,8 @@
-Track user operation on data models
-===================================
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+    :alt: License: AGPL-3
+
+Audit Log - Track user operations
```
Comment (elicoidal) [inline auditlog/README.rst:6]: you need to add one top line being document title

**Comment by elicoidal** [comment]
LGTM

File: auditlog/README.rst (hunk 8 above)
Comment (sebalix) [inline auditlog/README.rst:6]: Oh ok, fixed.

**Comment by sebalix** [comment]
The unit tests seems to fail sometimes, it is surely related to the date comparison in the search pattern... Do not merge yet, WIP

**Comment by sebalix** [comment]
Fixed, I added a `time.sleep(1)` in the unit test before the call to `autovacuum(days=0)` (https://github.com/OCA/server-tools/pull/360/files#diff-ca82e7cf6716c1ba4ff7d52084a0e14fR42).

File: auditlog/data/ir_cron.xml [hunk 9]
```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0" encoding="utf-8"?>
+<openerp>
+    <data noupdate="1">
+
+        <record id="ir_cron_auditlog_autovacuum" model="ir.cron">
+            <field name='name'>Auto-vacuum audit logs</field>
+            <field name='interval_number'>1</field>
+            <field name='interval_type'>days</field>
+            <field name="numbercall">-1</field>
+            <field name="active">True</field>
```
Comment (StefanRijnhart) [inline auditlog/data/ir_cron.xml]: Set this to False by default, so that it does not suprise any current users that their logs start to be wiped after half a year?

Comment (sebalix) [inline auditlog/data/ir_cron.xml]: Done, and README updated.

**Comment by sebalix** [comment]
2 thumbs up + more than 5 days, I think it can be merged? Someone available to do it? ;)

**Comment by pedrobaeza** [comment]
OK, merging.

**Comment by sebalix** [comment]
@pedrobaeza thank you!
